### DISCUSSION TITLE
Error "Update manager not supported" with debug configuration in BIRT 4.9

### DISCUSSION THREAD

**Discussion by NicolasPerritaz** [eclipse-birt/birt#907] [Q&A] [answered]
Hi guys,

I would like to launch debug configuration in BIRT 4.9, but I get the following error when I create a simple report and select "Run" -> "Debug Report".
I get the same error if I try to manually create a debug configuration of type "Report".

![image](https://user-images.githubusercontent.com/90317691/160666205-5ebeafff-9865-4240-abd0-e7a516ca849c.png)

Did anyone notice the same error? Should I create an issue?

**Comment by hvbtup** [comment]
As a side note, for me, "debug report" has never worked (with BIRT 3.7.1, 4.2.1, 4.3.0). To debug Javascript inside reports, I am using a Java logging library from Javascript.

**Reply by wimjongman** [reply]
I have tested it and it works nicely. You have to be aware of some things:

1. Add a breakpoint in any of the javascript code of the report
2. Choose Run/Debug Report
3. Open the debug view manually

Then it works like a charm. It stops on the correct line, you can inspect variable values and all that good stuff

**Comment by NicolasPerritaz** [comment]
For us it works with BIRT 4.6 as we use report debug configurations to debug java
We are not able to use it in BIRT 4.8, because the debug configuration doesn't start anything. I was hoping to be able to use it with BIRT 4.9, but I get this error message.

**Reply by wimjongman** [reply]
Can you please open the "Error Log View", export it, and attach it here?

**Reply by NicolasPerritaz** [reply]
Here it is.

[birt49.log](https://github.com/eclipse/birt/files/8381300/birt49.log)

**Comment by hvbtup** [comment]
This doesn't solve the issue, but may be a workaround:
You can debug the whole birt-all-in-one from the SDK IDE (see [Create a BIRT development environment](https://github.com/eclipse/birt#create-a-birt-development-environment)) and run your report from there. Then you should be able to e.g. use breakpoints in your Java code.

**Reply by NicolasPerritaz** [reply]
Thanks for the tip. I'll have a look. I think there would be too much overhead for a daily product usage, but it could be used to debug the issue itself I guess.

**Comment by pcornaz** [comment]
Although my comment doesn't help solving the issue, I still would like to stress the importance of this feature. Indeed, being able to extend BIRT's default behaviour with JavaScript and Java code is a key feature of the product; so, it's quite a downer that it doesn't work!

**Comment by pcornaz** [comment]
I did a quick search on this and based on https://git.eclipse.org/c/pde/eclipse.pde.ui.git/commit/?id=019fbb31a420e07e340e8ddbf81c5433ce7b58f0, it seems like one of the components (plug-ins, fragments, features, update sites and RCP products) expects the old update manager being present.  I'm not familiar enough with Eclipse PDE UI to make something out of this but maybe this will ring a bell with someone, and we can identify a workaround or even better a way to fix this in the next release... ?

**Comment by wimjongman** [comment, marked as answer]
I have fixed the issue. Please download the result from this job to test.

https://github.com/eclipse/birt/actions/runs/2066751417

**Reply by NicolasPerritaz** [reply]
I've tested the generated artifact and the debug configuration works!.

Many thanks Wim for this very important fix!

**Reply by wimjongman** [reply]
Thanks for the feedback.
